### PR TITLE
[ENG-413] Fetch email from buildkite env rather than commit author

### PR DESCRIPTION
**Pull request by jkburges**
Some devs have private emails in github, which results in a generated email address ending up in the git commit. This can't be used for looking people up in slack.

Instead, we use the buildkite email, which seems to be the dev's real email address.

### CODE REVIEW

**Comment by jkburges** [comment]
Hey @alviney @dominicwhittle, ENG-413 is about making deploys automatic. One of the things to facilitate that is direct pinging of people whose work is being deployed (since no longer need interaction with buildkite, might be easy to miss a deploy without direct pings).

Long story short the feature depends on buildkite email matching slack email but that is not the case for you two.

Viney, you are alex.viney@biteable.com in bk, alex@biteable.com in slack
Dom, you are dominic.whittle@biteable.com in bk, dom@weekends.ws in slack

How would both of you feel about changing so that they match? Alex, doesn't matter too much which way you go, but Dom, IMO it'd be preferable to go for the biteable.com address if possible.

**Comment by alviney** [comment]
> alex@biteable.com

Yeah that's cool. I've changed my bk primary to alex@biteable.com

File: lib/lookup_user.bash [hunk 1]
```diff
@@ -1,7 +1,6 @@
 #!/bin/bash
 lookupUser() {
-    local email=`git log -1 --format='%ae'`
-    local user_id=`curl -s -H "Authorization: Bearer $BUILDKITE_BOT_TOKEN" "https://slack.com/api/users.lookupByEmail?email=$email" | sed -E 's/.*"id":"(\w+)".*/\1/g'`
+    local user_id=`curl -s -H "Authorization: Bearer $BUILDKITE_BOT_TOKEN" "https://slack.com/api/users.lookupByEmail?email=$BUILDKITE_BUILD_AUTHOR_EMAIL" | sed -E 's/.*"id":"(\w+)".*/\1/g'`
```
Comment (wbowling) [inline lib/lookup_user.bash:3]: is it worth adding both here if one isn't found? Otherwise just fixing peoples emails when it errors is probably fine too

**Comment by dominicwhittle** [comment]
I can't see where I can change my Slack email, but have added dom@weekends.ws (in addition to the existing dominic.whittle@biteable.com) in Buildkite.

File: lib/lookup_user.bash (hunk 1 above)
Comment (jkburges) [reply lib/lookup_user.bash:3]: Both BUILDKITE_BUILD_AUTHOR_EMAIL and one from the commit, do you mean?

**Comment by jkburges** [comment]
> I can't see where I can change my Slack email, but have added [dom@weekends.ws](mailto:dom@weekends.ws) (in addition to the existing [dominic.whittle@biteable.com](mailto:dominic.whittle@biteable.com)) in Buildkite.

I think that should work as `BUILDKITE_BUILD_AUTHOR_EMAIL="dom@weekends.ws"` AFAICT e.g. https://buildkite.com/biteable/biteable-www/builds/201#77a37cb9-d123-4f7a-8ffe-8355850aa1ee

In case you did want to change your slack email though, https://biteable.slack.com/account/settings is where you do it (it's hard to find).

File: lib/lookup_user.bash (hunk 1 above)
Comment (wbowling) [reply lib/lookup_user.bash:3]: yeah, but I think how it is now is good

**Review by wbowling — APPROVED** [review]
:shipit:

**Pull request merged**